scripts/backupmanager.sh

```diff
@@ -0,0 +1,336 @@
+            if [ ${#} -ne 1 ]; then
+                pp --error "Invalid number of arguments for 'create'\n"
+                print_usage
+                exit 1
+            fi
+            create_backup
+            ;;
+        -l|--list)
+            if [ ${#} -gt 2 ]; then
+                pp --error "Invalid number of arguments for 'list'\n"
+                print_usage
+                exit 1
+            fi
+            list_backups "${2:-}"
+            ;;
+        # -r|--restore)
+        #     if [ ${#} -gt 2 ]; then
+        #         pp --error "Invalid number of arguments for 'restore'\n"
+        #         print_usage
+        #         exit 1
+        #     fi
+        #     restore_backup "${2:-}"
+        #     ;;
+        -cl|--clean)
+            if [ ${#} -gt 2 ]; then
+                pp --error "Invalid number of arguments for 'clean'\n"
+                print_usage
+                exit 1
+            fi
+            clean_backups "${2:-}"
+            ;;
+        -h|--help)
+            if [ ${#} -ne 1 ]; then
+                pp --error "Invalid number of arguments for 'help'\n"
+                print_usage
+                exit 1
+            fi
+            print_usage
+            ;;
+        *)
+            pp --error "Illegal option '${1}'\n"
+            print_usage
+            exit 1
+            ;;
+    esac
+}
+
+function check_required_directories() {
+    # Exit if backup dir or game save dir don't exist
+    # Check if env vars set as default to local vars exist
+    if [ -z "${BACKUP_DIR}" ]; then
+        pp --error "> BACKUP_PATH environment variable not set.\n"
+        pp --error "> Please provide the backup directory as an argument or set the BACKUP_PATH environment variable.\n"
+        pp --error "> Exiting...\n"
+        exit 1
+    elif [ -z "${SAVE_DIR}" ]; then
+        pp --error "> GAME_SAVE_PATH environment variable not set.\n"
+        pp --error "> Exiting...\n"
+        exit 1
+    fi
+
+    # If vars are set, check if the directories exist
+    if [ ! -d "${BACKUP_DIR}" ]; then
+        pp --error "[ERROR] Backup directory '${BACKUP_DIR}' does not exist.\n"
+        exit 1
+    elif [ ! -d "${SAVE_DIR}" ]; then
+        pp --error "[ERROR] Game save directory '${SAVE_DIR}' does not exist.\n"
+        exit 1
+    fi
+}
+
+
+### Backup Functions
+
+function create_backup() {
+
+    pp --success ">>> Creating backup...\n\n"
+
+
+    if [ -z "${GAME_PATH}" ]; then
+        pp --error "> RESTORE_FILE_PATH environment variable not set. Exiting...\n"
+        exit 1
+    fi
+    
+    DATE=$(date +%Y%m%d_%H%M%S)
+    TIME=$(date +%H-%M-%S)
+
+    #Send message to gameserver via RCON if enabled
+    rc "broadcast ${TIME}-Backup_in_progress" "> Broadcasting server shutdown via RCON...\n"
+
+    # Create backup dir if it doesn't exist
+    mkdir -p "${BACKUP_DIR}"
+
+    #Send message to gameserver via RCON if enabled
+    rc 'broadcast Saving world...' "> Sending message to gameserver via RCON..."
```

Comment:
This wont work
broadcast cant have " ", meaning spaces, there is a reason i wrote "stuff-likes-this-to-the-gameserver".